share/src/main/java/org/apache/kafka/server/share/fetch/ShareFetch.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.server.share.fetch;
+
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.common.message.ShareFetchResponseData.PartitionData;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.server.storage.log.FetchParams;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * The ShareFetch class is used to store the fetch parameters for a share fetch request.
+ */
+public class ShareFetch {
+
+    /**
+     * The future that will be completed when the fetch is done.
+     */
+    private final CompletableFuture<Map<TopicIdPartition, PartitionData>> future;
+
+    /**
+     * The fetch parameters for the fetch request.
+     */
+    private final FetchParams fetchParams;
+    /**
+     * The group id of the share group that is fetching the records.
+     */
+    private final String groupId;
+    /**
+     * The member id of the share group that is fetching the records.
+     */
+    private final String memberId;
+    /**
+     * The maximum number of bytes that can be fetched for each partition.
+     */
+    private final Map<TopicIdPartition, Integer> partitionMaxBytes;
+    /**
+     * The maximum number of records that can be fetched for the request.
+     */
+    private final int maxFetchRecords;
+    /**
+     * The partitions that had an error during the fetch.
+     */
+    private volatile Map<TopicIdPartition, Throwable> erroneous;
```

Comment:
`volatile` guarantees that a subsequent reader will pick up the latest reference to the map, but not necessarily the latest content in the map.

Reply:
I have moved errorneous in synchronized, thanks for pointing out.